src/portal/lib/src/create-edit-rule/create-edit-rule.component.html

```diff
@@ -68,7 +68,7 @@ <h3 class="modal-title">{{headerTitle | translate}}</h3>
                 </div>
                 <label *ngIf="supportedFilters[i]?.style==='input'" aria-haspopup="true" role="tooltip" class="tooltip tooltip-validation tooltip-md tooltip-bottom-left"
                   [class.invalid]='(filter.value.dirty || filter.value.touched) && filter.value.invalid'>
-                  <input type="text" #filterValue required size="14" formControlName="value" id="{{'filter_'+ supportedFilters[i]?.type}}">
+                  <input (input)="trimText($event)" type="text" #filterValue required size="14" formControlName="value" id="{{'filter_'+ supportedFilters[i]?.type}}">
```

Comment:
Why need trim() here. not in the form.

Reply:
the form is dynamic .  it will be much more complicated to use trim in the form.